### PR TITLE
Rust, React, Raspberry Pi

### PR DESCRIPTION
**Pull request by andricDu**

### CODE REVIEW

File: _posts/dusan_andric/2020-12-11-bombolone.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+## Hardware
+
+I had been looking for an excuse to play with a Raspberry Pi for a long time and now finally I had one. The Raspberry Pi 4 had launched the previous year, and with beefier specs compared the previous gen, it looked like a great candidate for being a dev test bench. Also, most importantly, it drew power from a USB C connector!
+
+I ordered my RPi4 kit from [CanaKit](https://www.canakit.com/) and as time went on I added components that I ordered from [ByAPi.ca](https://www.buyapi.ca/).
+
+Knowing that night vision capabilities were essential, I made sure to order a Pi Camera that did not have an infrared filter as well as some IR emitting diodes.  
+
+Once I connected it all together, I validate the functionality of the night vision with a simple python script that utilized the picamera API to serve a mjpg stream over HTTP. You can find an example on how to do this [here](https://picamera.readthedocs.io/en/release-1.13/recipes2.html#web-streaming).
```
Comment (rosibaj) [inline _posts/dusan_andric/2020-12-11-bombolone.md]: Once I connected it all together, I `validated` the functionality....

File: _posts/dusan_andric/2020-12-11-bombolone.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/bombolone/pi-connected.png" />
+    <figcaption>Power over USB C, display via HDMI, and Wireless USB dongle for keyboard/mouse.</figcaption>
+</figure>
+
+## Software
+
+Now that I had the hardware assembled, I wanted to see how far I could stretch the python script. I left the script running, accessibly by local network for a time. I quickly ran into stability issues. The HTTP access to the stream would become unresponsive when multiple clients would be left streaming, for example, the phones of two ever concerned parents. The python process itself also seemed to leak memory over time, not that was an issue with 4GB of RAM on the RPi4.
```
Comment (rosibaj) [inline _posts/dusan_andric/2020-12-11-bombolone.md]: I left the script running, `accessible` by local network for a time.

**Review by rosibaj — CHANGES_REQUESTED** [review]

**Review by rosibaj — APPROVED** [review]

File: _posts/dusan_andric/2020-12-11-bombolone.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        this.state.showQr
+        ? <QrReader delay={300} onError={(err) => this.handleError(err)} onScan={(data) => this.handleScan(data)} style={{ 'width': '100%', 'margin-right': '-50%' }}/>
+        : <button className="ui button" onClick={() => this.showQr()}>Scan QR Code</button>
+    }
+</div>
+```
+
+## Ship It!
+
+With the main software components ready, it was time to handle the infrastructure. I needed to figure out how to make my baby monitor reachable form the internet for when I am out the house. I already own a domain, but the real trick is how do I solve the issue of being behind a carrier grade NAT.
```
Comment (denis-yuen) [inline _posts/dusan_andric/2020-12-11-bombolone.md:147]: form -> from, out -> out of

File: _posts/dusan_andric/2020-12-11-bombolone.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    <img src="{{site.urlimg}}dusan_andric/bombolone/usb-b.png" />
+    <figcaption>The whole Micro USB B port ended up ripping off of the PCB!</figcaption>
+</figure>
+</center>
+
+## Hardware
+
+I had been looking for an excuse to play with a Raspberry Pi for a long time and now finally I had one. The Raspberry Pi 4 had launched the previous year, and with beefier specs compared the previous gen, it looked like a great candidate for being a dev test bench. Also, most importantly, it drew power from a USB C connector!
+
+I ordered my RPi4 kit from [CanaKit](https://www.canakit.com/) and as time went on I added components that I ordered from [ByAPi.ca](https://www.buyapi.ca/).
```
Comment (denis-yuen) [inline _posts/dusan_andric/2020-12-11-bombolone.md:51]: BuyAPi -> buyapi.ca

**Review by denis-yuen — COMMENTED** [review]
Might be too late since this is merged, but a cool read and will check out buyapi.ca

Edit: also super late

**Pull request merged**